src/NUnitCore/core/Compatibility.cs

```diff
@@ -0,0 +1,298 @@
+
+            foreach (Type type in assembly.GetExportedTypes())
+            {
+                CheckAttributes(type, type.FullName);
+
+                foreach (MemberInfo member in type.GetMembers())
+                {
+                    CheckAttributes(member);
+                    var method = member as MethodInfo;
+                    if (method != null)
+                        foreach (var parameter in method.GetParameters())
+                            CheckAttributes(parameter, LocationOf(method));
+                }
+            }
+        }
+
+        public static void CheckAttributes(MemberInfo member)
+        {
+            CheckAttributes(member, LocationOf(member));
+        }
+
+        public static void CheckAttributes(ICustomAttributeProvider provider, string location)
+        {
+            var methodInfo = provider as MethodInfo;
+            var parameterInfo = provider as ParameterInfo;
+
+            foreach (Attribute attribute in provider.GetCustomAttributes(true))
+            {
+                Type attributeType = attribute.GetType();
+                string attributeName = attributeType.Name;
+                string attributeFullName = attributeType.FullName;
+
+                switch (attributeFullName)
+                {
+                    case "NUnit.Framework.ExpectedExceptionAttribute":
+                        Error(location, "ExpectedExceptionAttribute is not supported in NUnit 3. Use Assert.Throws or ThrowsConstraint.");
+                        break;
+                    case "NUnit.Framework.IgnoreAttribute":
+                        var reason = (string)Reflect.GetPropertyValue(attribute, "Reason");
+                        if (string.IsNullOrEmpty(reason))
+                            Error(location, "IgnoreAttribute must have a reason specified in NUnit 3.");
+                        break;
+                    case "NUnit.Framework.RequiresSTAAttribute":
+                    case "NUnit.Framework.RequiresMTAAttribute":
+                        Error(location, attributeName + " is not supported in NUnit 3. Use ApartmentAttribute.");
+                        break;
+                    case "NUnit.Core.Extensibility.NUnitAddinAttribute":
+                    case "NUnit.Framework.RequiredAddinAttribute":
+                        Error(location, attributeName + " is not available in NUnit 3, which no longer supports Addins. After conversion, you can create custom attributes or engine extensions instead.");
+                        break;
+                    case "NUnit.Framework.SuiteAttribute":
+                        Error(location, "SuiteAttribute is not supported in NUnit 3. You should restructure your tests to eliminate legacy Suites.");
+                        break;
+                    case "NUnit.Framework.SetUpAttribute":
+                    case "NUnit.Framework.TearDownAttribute":
+                        if (methodInfo != null && methodInfo.ReflectedType != null &&
```

Comment:
Option: `methodInfo?.ReflectedType != null`

Reply:
Nope... won't compile in the batch build. See my earlier comment. FWIW, it's a real pain to work on this old code.